metadata/src/main/java/org/apache/kafka/image/loader/MetadataLoader.java

```diff
@@ -347,7 +347,7 @@ private void maybePublishMetadata(MetadataDelta delta, MetadataImage image, Load
         }
         metrics.updateLastAppliedImageProvenance(image.provenance());
         metrics.setCurrentMetadataVersion(image.features().metadataVersion());
-        if (uninitializedPublishers.isEmpty()) {
+        if (!uninitializedPublishers.isEmpty()) {
```

Comment:
Nice catch

Reply:
Sigh. Looks like this was introduced in KAFKA-14538, which I reviewed. It really only bites in scenarios where you're running a pre-NO_OP record MV, which is rare these days.

Thanks for the fix.